cmd/spire-agent/cli/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,103 @@
+package healthcheck
+
+import (
+	"errors"
+	"flag"
+	"net"
+	"time"
+
+	"github.com/mitchellh/cli"
+	api_workload "github.com/spiffe/spire/api/workload"
+	"github.com/spiffe/spire/cmd/spire-agent/cli/common"
+	common_cli "github.com/spiffe/spire/pkg/common/cli"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func NewHealthCheckCommand() cli.Command {
+	return newHealthCheckCommand(common_cli.DefaultEnv)
+}
+
+func newHealthCheckCommand(env *common_cli.Env) *healthCheckCommand {
+	return &healthCheckCommand{
+		env:     env,
+		timeout: common_cli.DurationFlag(time.Second * 5),
+	}
+}
+
+type healthCheckCommand struct {
+	env *common_cli.Env
+
+	socketPath string
+	timeout    common_cli.DurationFlag
+	shallow    bool
```

Comment:
is shallow not current being used?

Reply:
Nope. The only reason I put it there is so that the flag can be passed now for liveliness checking and if/when we define different semantics for a shallow test, those changes will be picked up w/o having to mess with config.

Reply:
Maybe it's better to leave it out and force people to explicitly opt in when the shallow check actually gets implemented.